.github/workflows/ci.yaml

```diff
@@ -65,30 +77,158 @@ jobs:
+          if [[ "$BUMP_MANIFEST" == "true" ]]; then
+            # This branch is for scheduled nightlies or manually run nightlies
+            MANIFEST_BRANCH=znightly-${{ steps.date.outputs.BUILD_DATE }}-${{ github.run_id }}
+            MANIFEST_ARTIFACT_NAME=${{ env.DEFAULT_MANIFEST_ARTIFACT_NAME }}
+          else
+            # This branch is for presubmits (no bumping needed)
+            MANIFEST_BRANCH=${{ github.sha }}
+            # Empty artifact name means to use the one in version control
+            MANIFEST_ARTIFACT_NAME=""
+          fi
+          echo "MANIFEST_BRANCH=$MANIFEST_BRANCH" | tee -a $GITHUB_OUTPUT
+          echo "MANIFEST_ARTIFACT_NAME=$MANIFEST_ARTIFACT_NAME" | tee -a $GITHUB_OUTPUT
+          echo "BUMP_MANIFEST=$BUMP_MANIFEST" | tee -a $GITHUB_OUTPUT
+          echo "MERGE_BUMBED_MANIFEST=$MERGE_BUMPED_MANIFEST" | tee -a $GITHUB_OUTPUT
+          if [[ "$BUMP_MANIFEST" == "false" && "$MERGE_BUMPED_MANIFEST" == "true" ]]; then
+            echo "Error: If BUMP_MANIFEST=false, MERGE_BUMPED_MANIFEST cannot be true" >&2
+            exit 1
+          fi
 
-  amd64:
+  maybe-bump-manifest:
     needs: metadata
+    runs-on: ubuntu-22.04
+    if: needs.metadata.outputs.BUMP_MANIFEST == 'true'
+    steps:
+      - name: Check out the repository under ${GITHUB_WORKSPACE}
+        uses: actions/checkout@v4
+
+      - name: Update manifest and patches in-place - show diff
+        working-directory: .github/container
+        shell: bash -x -e {0}
+        run: |
+          bash bump.sh --input-manifest manifest.yaml
+          git diff
+      
+      - name: Upload bumped manifest/patches to be used in build-base
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ needs.metadata.outputs.MANIFEST_ARTIFACT_NAME }}
+          path: |
+            .github/container/manifest.yaml
+            .github/container/patches
+
+  amd64:
+    needs: [metadata, maybe-bump-manifest]
+    if: "!cancelled()"
     uses: ./.github/workflows/_ci.yaml
     with:
       ARCHITECTURE: amd64
       BUILD_DATE: ${{ needs.metadata.outputs.BUILD_DATE }}
-      BUMP_MANIFEST: ${{ needs.metadata.outputs.BUMP_MANIFEST == 'true' }}
+      MANIFEST_ARTIFACT_NAME: ${{ needs.metadata.outputs.MANIFEST_ARTIFACT_NAME }}
     secrets: inherit
 
   arm64:
-    needs: metadata
+    needs: [metadata, maybe-bump-manifest]
+    if: "!cancelled()"
     uses: ./.github/workflows/_ci.yaml
     with:
       ARCHITECTURE: arm64
       BUILD_DATE: ${{ needs.metadata.outputs.BUILD_DATE }}
-      BUMP_MANIFEST: ${{ needs.metadata.outputs.BUMP_MANIFEST == 'true' }}      
+      MANIFEST_ARTIFACT_NAME: ${{ needs.metadata.outputs.MANIFEST_ARTIFACT_NAME }}
     secrets: inherit
 
+  # Only merge if everything succeeds
+  merge-new-manifest:
+    runs-on: ubuntu-22.04
+    if: ${{ !cancelled() && needs.metadata.outputs.MERGE_BUMPED_MANIFEST == 'true' && needs.metadata.outputs.MANIFEST_BRANCH != github.sha }}
```

Comment:
Would it make better same to remove the `!cancelled()` part so that we can cancel a workflow as a 'emergency-stop' button to stop merging?

Reply:
My understanding is I need this `!cancelled()` b/c I want `mege-new-manifest` to run regardless of the `needs` jobs succeeding. If any of them fail, then there's an implicit "skip descendants" logic so if arm64 failed, I would skip this entire job. The reason I want the job to run regardless is so I can create a PR.

Also, doesn't `!cancelled()` respect emergency stop? Isn't that why it's preferred over `always()`?

Reply:
> Also, doesn't !cancelled() respect emergency stop? Isn't that why it's preferred over always()?

You are right 😂 

> I want mege-new-manifest to run regardless of the needs jobs succeeding.

Is this supposedly the 'all-green' merge so that this would not run if the whole run is not successful? Or is it because you want the PR to be created anyway, and just not *merge* it if it does not succeed?

Reply:
It would be fine for the time being, but in the long run I would rather we do not create PRs from failed runs. Instead, (as we discussed offline last week), we should create a separate workflow with a manual `workflow_dispatch` trigger to turn a workflow run into a PR.